anchore_engine/services/policy_engine/engine/vulns/providers.py

```diff
@@ -782,14 +798,35 @@ def get_image_vulnerabilities(
         use_store: bool = True,
     ) -> ImageVulnerabilitiesReport:
         if force_refresh:
-            return self._create_new_report(image, db_session, use_store)
+            report = self._create_new_report(image, db_session, use_store)
         else:
             report = self._try_load_report_from_store(image, db_session, use_store)
-            if isinstance(report, ImageVulnerabilitiesReport):
-                return report
-            else:
-                report = ImageVulnerabilitiesReport.from_json(report)
-                return report
+
+        if not isinstance(report, ImageVulnerabilitiesReport):
```

Comment:
why not check that it's a dict instead of checking that it's not a ImageVulnerabilitiesReport

Reply:
It's the same result at the end, this flow made more sense in my head: if it's not a report object, then create one first before operating on it